os/drivers/serial/serial.c

```diff
@@ -897,6 +897,14 @@ static int uart_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 		}
 		break;
 
+		case FIOC_TXEMPTY: {
```

Comment:
there was an requirement in product level, they wanted to check not only empty but ready status too.
So I added code like a below
```
	case TIOCTXSTATE: {
		uint16_t *status = (uint16_t *)((uintptr_t)arg);

		*status = TIOCS_NONE;
        if ((dev->xmit.head == dev->xmit.tail) && uart_txempty(dev)) {
            *status |= TIOCS_EMPTY;
        }
 
        if (uart_txready(dev)) {
            *status |= TIOCS_READY;
        }

	}
```
you can refer to micom_controller in our tp1x repo.

Reply:
done